crates/rpc/rpc-builder/src/eth.rs

```diff
@@ -1,232 +1,136 @@
-        )
-    }
-
-    /// Initializes the `FeeHistoryCache`.
-    fn init_fee_history_cache(&self, cache: &EthStateCache) -> FeeHistoryCache {
-        FeeHistoryCache::new(cache.clone(), self.rpc_config.eth.fee_history_cache.clone())
-    }
-
-    /// Spawns background tasks for updating caches.
-    fn spawn_cache_tasks(&self, cache: &EthStateCache, fee_history_cache: &FeeHistoryCache) {
-        // Get the stream of new canonical blocks
-        let new_canonical_blocks = self.eth_handlers_config.events.canonical_state_stream();
-
-        // Clone the cache for the task
-        let cache_clone = cache.clone();
+    /// Returns a new instance with handlers for `eth` namespace.
+    pub fn build(self) -> EthHandlers<Provider, Pool, Network, Events, EthApi> {
+        let Self { provider, pool, network, evm_config, config, executor, events, eth_api_builder } =
+            self;
+
+        let cache = EthStateCache::spawn_with(
+            provider.clone(),
+            config.cache,
+            executor.clone(),
+            evm_config.clone(),
+        );
 
-        // Spawn a critical task to update the cache with new blocks
-        self.eth_handlers_config.executor.spawn_critical(
+        let new_canonical_blocks = events.canonical_state_stream();
+        let c = cache.clone();
+        executor.spawn_critical(
             "cache canonical blocks task",
             Box::pin(async move {
-                cache_new_blocks_task(cache_clone, new_canonical_blocks).await;
+                cache_new_blocks_task(c, new_canonical_blocks).await;
             }),
         );
 
-        // Get another stream of new canonical blocks
-        let new_canonical_blocks = self.eth_handlers_config.events.canonical_state_stream();
-
-        // Clone the fee history cache for the task
-        let fhc_clone = fee_history_cache.clone();
+        let ctx = EthApiBuilderCtx {
+            provider,
+            pool,
+            network,
+            evm_config,
+            config,
+            executor,
+            events,
+            cache,
+        };
 
-        // Clone the provider for the task
-        let provider_clone = self.eth_handlers_config.provider.clone();
-
-        // Spawn a critical task to update the fee history cache with new blocks
-        self.eth_handlers_config.executor.spawn_critical(
-            "cache canonical blocks for fee history task",
-            Box::pin(async move {
-                fee_history_cache_new_blocks_task(fhc_clone, new_canonical_blocks, provider_clone)
-                    .await;
-            }),
-        );
-    }
+        let api = eth_api_builder(&ctx);
 
-    /// Initializes the `GasPriceOracle`.
-    fn init_gas_oracle(&self, cache: &EthStateCache) -> GasPriceOracle<Provider> {
-        GasPriceOracle::new(
-            self.eth_handlers_config.provider.clone(),
-            self.rpc_config.eth.gas_oracle.clone(),
-            cache.clone(),
-        )
-    }
```

Comment:
did we replace these functions with builder structs?
why?

Reply:
in anticipation that we will make those apis configurable in the future too, exposing all eth handlers to user (and I had already implemented it before Thomas pr was merged)